apps/swirl-docs/src/components/Components/ComponentGrid.tsx

```diff
@@ -0,0 +1,65 @@
+import useDynamicRefs, {
+  handleGridKeyDown,
+} from "@swirl/lib/hooks/useDynamicRefs";
+import { FunctionComponent, LegacyRef } from "react";
+import Grid from "src/components/Grid";
+import { NavItem } from "@swirl/lib/navigation";
+import Image from "next/image";
+
+interface ComponentGridProps {
+  componentList: NavItem[];
+}
+
+export const ComponentGrid: FunctionComponent<ComponentGridProps> = ({
+  componentList,
+}) => {
+  const [getRef, setRef] = useDynamicRefs();
+
+  return (
+    <Grid className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-2 xl:grid-cols-3 gap-4 w-full">
+      {componentList?.map((component: NavItem, index: number) => (
+        <a
+          key={index}
+          role="gridcell"
```

Comment:
**issue:** `role="gridcell"` is not allowed on an `a` tag. You could move the role and label to the article.